### PR TITLE
ENH: Add indent to image's matrix printing

### PR DESCRIPTION
**Pull request by blowekamp**
Add Matrix::PrintSelf method which uses ITK's indent object to allow proper indenting of nested objects.

Fixes left aligned matrix values when printing the Image classes.

## PR Checklist
- [ ] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [ ] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)
- [ ] Added test (or behavior not changed)
- [ ] Updated API documentation (or API not changed)
- [ ] Added [license](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Utilities/KWStyle/ITKHeader.h) to new files (if any)
- [ ] Added Python wrapping to new files (if any) as described in [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) Section 9.5
- [ ] Added [ITK examples](https://github.com/InsightSoftwareConsortium/ITKSphinxExamples) for all new major features (if any)

Refer to the [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) for
further development details if necessary.

### CODE REVIEW

**Review by dzenanz — APPROVED** [review]

**Comment by blowekamp** [comment]
This was the print out style that Copilot made up. I am not sure if a numpy style with `[[][][]]` would be preferred, or the the generated matrix dimensions are desired. Please suggest/propose.

New printout from Image:
```
603:   Direction: 
603:   Matrix (2x2)
603:     1 0 
603:     0 1 
603:   IndexToPointMatrix: 
603:     Matrix (2x2)
603:       1 0 
603:       0 1 
603:   PointToIndexMatrix: 
603:     Matrix (2x2)
603:       1 0 
603:       0 1 
603:   Inverse Direction: 
603:     Matrix (2x2)
603:       1 0 
603:       0 1 

```

Old print out:
```

  Direction: 
1 0
0 1

  IndexToPointMatrix: 
1 0
0 1

  PointToIndexMatrix: 
1 0
0 1

  Inverse Direction: 
1 0
0 1

```

And a numpy identity matrix:
```
[[1. 0. 0.]
 [0. 1. 0.]
 [0. 0. 1.]]
```

**Comment by dzenanz** [comment]
I like it this way. I thought about whether to ask for before/after output :smile:

File: Modules/Core/Common/include/itkImageBase.hxx [hunk 1]
```diff
@@ -528,16 +528,18 @@ ImageBase<VImageDimension>::PrintSelf(std::ostream & os, Indent indent) const
 
   os << indent << "Origin: " << this->GetOrigin() << std::endl;
 
-  os << indent << "Direction: " << std::endl << this->GetDirection() << std::endl;
+
+  os << indent << "Direction: " << std::endl;
+  this->GetDirection().PrintSelf(os, indent);
```
Comment (blowekamp) [inline Modules/Core/Common/include/itkImageBase.hxx:533]: This should probably be `indent` -> `indent.GetNextIndent()`

Comment (jhlegarreta) [reply Modules/Core/Common/include/itkImageBase.hxx:533]: I think so.

**Review by jhlegarreta — APPROVED** [review]
Given the choice, maybe because I got used to it, I like Numpy's better (with the appropriate indentation, but I am fine with the current style improvement.

**Comment by blowekamp** [comment]
Note: the matrix in AffineTransform is printed correctly with in a CompositeTransform. The Image class should be consistent.

**Comment by blowekamp** [comment]
Let see if @thewtex has a comment about the preferred formatting of the output for the Matrix.

**Review by thewtex — APPROVED** [review]
:+1: looks like a good improvement to me. @blowekamp thanks for taking the time to make the tweak.

**Comment by dzenanz** [comment]
@blowekamp time to merge?

**Pull request merged**